velox/functions/prestosql/DateTimeImpl.h

```diff
@@ -60,4 +63,124 @@ FOLLY_ALWAYS_INLINE std::optional<Timestamp> fromUnixtime(double unixtime) {
   auto nanos = unixtime - seconds;
   return Timestamp(seconds, nanos * kNanosecondsInSecond);
 }
+
+namespace {
+enum class DateTimeUnit {
+  kMillisecond,
+  kSecond,
+  kMinute,
+  kHour,
+  kDay,
+  kMonth,
+  kQuarter,
+  kYear
+};
+} // namespace
+
+// Year, quarter or month are not uniformly incremented. Months have different
+// total days, and leap years have more days than the rest. If the new year,
+// quarter or month has less total days than the given one, it will be coerced
+// to use the valid last day of the new month. For example,
+//
+// 2022-01-30 + (1 month) = 2022-02-28
+// 2022-02-28 - (1 month) = 2022-01-28
+//
+// 2022-08-31 + (1 quarter) = 2022-11-30
+// 2022-11-30 - (1 quarter) = 2022-08-30
+//
+// 2020-02-29 + (1 year) = 2021-02-28
+// 2021-02-28 - (1 year) = 2020-02-28
+FOLLY_ALWAYS_INLINE Date
```

Comment:
Please can you change the formating to have the returntype Date on the same line as the function name to make it more readable.